llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -2529,6 +2529,23 @@ static SDValue foldAddSubBoolOfMaskedVal(SDNode *N, SelectionDAG &DAG) {
   return DAG.getNode(IsAdd ? ISD::SUB : ISD::ADD, DL, VT, C1, LowBit);
 }
 
+// Attempt to form avgceilu(A, B) from (A | B) - ((A ^ B) >> 1)
+static SDValue combineFixedwidthToAVGCEILU(SDNode *N, SelectionDAG &DAG) {
+  const TargetLowering &TLI = DAG.getTargetLoweringInfo();
+  SDValue N0 = N->getOperand(0);
+  EVT VT = N0.getValueType();
+  SDLoc DL(N);
+  if (TLI.isOperationLegal(ISD::AVGFLOORU, VT)) {
```

Comment:
AVGFLOORU -> AVGCEILU

This shares quite a lot with #84903 and could be combined into the same function once that is submitted.

Reply:
> AVGFLOORU -> AVGCEILU

Updated.

> This shares quite a lot with https://github.com/llvm/llvm-project/pull/84903 and could be combined into the same function once that is submitted

Sure.